### PR TITLE
Added macOS compile script

### PR DESCRIPTION
**Pull request by kms1212**
According to issue #67, I added a script for compiling binary for macOS.

### CODE REVIEW

**Comment by CATboardBETA** [comment]
This does not function on M1 MacBook Pro 2021. I've disabled SIP and given terminal full disk access, but I still get "Operation not permitted" when I run macOS-compile.sh.

**Comment by olback** [comment]
> This does not function on M1 MacBook Pro 2021. I've disabled SIP and given terminal full disk access, but I still get "Operation not permitted" when I run macOS-compile.sh.

Did you make the script executable? `chmod +x macOS-compile.sh.`

**Comment by CATboardBETA** [comment]
Yes, I did that previously

**Comment by cvpeck** [comment]
additional info for MacOS on Intel.

`error[E0463]: can't find crate for `core`
  |
  = note: the `aarch64-apple-darwin` target may not be installed
  = help: consider downloading the target with `rustup target add aarch64-apple-darwin`
`

Probably worth adding that to the instructions

**Comment by cvpeck** [comment]
further error (not sure how to fix)
`
The following warnings were emitted during compilation:

warning: pkg-config has not been configured to support cross-compilation.

error: failed to run custom build command for `glib-sys v0.14.0`

Caused by:
  process didn't exit successfully: `/Users/xxxx/Documents/KiCad/library-loader/target/release/build/glib-sys-0bad67eb640f0555/build-script-build` (exit status: 1)
  --- stdout
  cargo:rerun-if-env-changed=GLIB_2.0_NO_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG_ALLOW_CROSS_aarch64-apple-darwin
  cargo:rerun-if-env-changed=PKG_CONFIG_ALLOW_CROSS_aarch64_apple_darwin
  cargo:rerun-if-env-changed=TARGET_PKG_CONFIG_ALLOW_CROSS
  cargo:rerun-if-env-changed=PKG_CONFIG_ALLOW_CROSS
  cargo:rerun-if-env-changed=PKG_CONFIG_aarch64-apple-darwin
  cargo:rerun-if-env-changed=PKG_CONFIG_aarch64_apple_darwin
  cargo:rerun-if-env-changed=TARGET_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_aarch64-apple-darwin
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_aarch64_apple_darwin
  cargo:rerun-if-env-changed=TARGET_PKG_CONFIG_SYSROOT_DIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR
  cargo:warning=pkg-config has not been configured to support cross-compilation.

  Install a sysroot for the target platform and configure it via
  PKG_CONFIG_SYSROOT_DIR and PKG_CONFIG_PATH, or install a
  cross-compiling wrapper for pkg-config and set it via
  PKG_CONFIG environment variable.
warning: build failed, waiting for other jobs to finish...
error: build failed
`

This is on MacOS Monterey 12.1 Intel

**Comment by kms1212** [comment]
@cvpeck I think I set the cargo build target to only the Apple M1 chip. 
I'm sorry but please edit `macos-compile.sh` script and run it.

From:
<pre><code>$CARGO build --release --target=aarch64-apple-darwin</pre></code>

To:
<pre><code>$CARGO build --release --target=x86_64-apple-darwin</pre></code>

This might be working/

**Comment by kms1212** [comment]
@cvpeck
I tried to build on x86_64 with Rosetta, but the same problem has occurred. glib-sys doesn't detect glib package from pkg-config on x86_64, I'm sorry but I cannot fix this problem. To prevent confusion, I renamed `macos-compile.sh` to `macos-compile-aarch64.sh`.

**Comment by cvpeck** [comment]
Thanks, I successfully built on x86_64 by modifying the the compile script as suggested.
Due to caching issues I had to delete local copies then repull repo to ensure a clean build.

**Comment by kms1212** [comment]
Thank you for testing the script!
I'll restore the macOS build script for x86_64.

**Comment by olback** [comment]
You shouldn't need to specify the target when running/compiling. The compiler know what architecture it is running on and will target that automatically.

Is this ready to merge?

File: README.md [hunk 1]
```diff
@@ -18,15 +18,28 @@ Mac | WIP
...
 docker run --volume=$(pwd):/home/circleci/project olback/rust-gtk-linux cargo build --release
 ```
 
+### Building from source locally(Apple Sillicon macOS)
+
+Required binaries: brew(from homebrew), rustc, cargo
+You have to install rust via rustup and initialize it with rustup-init command.
+
+```shell
+./macos-compile-aarch64.sh
```
Comment (olback) [inline README.md:34]: Shouldn't this be `./macos-compile.sh`?

Comment (kms1212) [reply README.md]: Sorry, it's my mistake.

**Pull request merged**